pkg/tenant/auth_test.go

```diff
@@ -36,39 +36,69 @@ func (s *tenantTestSuite) TestRegister() {
 		Created: time.Now().Format(time.RFC3339Nano),
 	}
 
-	// Configure the initial mock to return a 200 response with the reply
-	s.quarterdeck.OnRegister(mock.UseStatus(http.StatusOK), mock.UseJSONFixture(reply))
+	// Configure the initial mock to return a 201 response with the reply
+	s.quarterdeck.OnRegister(mock.UseStatus(http.StatusCreated), mock.UseJSONFixture(reply))
 
-	// Name is required
+	// Test missing fields
 	req := &api.RegisterRequest{
-		Email:    "leopold.wentzel@gmail.com",
-		Password: "hunter2",
-		PwCheck:  "hunter2",
+		Name:         "Leopold Wentzel",
+		Email:        "leopold.wentzel@gmail.com",
+		Password:     "hunter2",
+		PwCheck:      "hunter2",
+		Organization: "Rotational Labs",
+		Domain:       "rotational.io",
+		AgreeToS:     true,
+		AgreePrivacy: true,
+	}
+	testCases := []struct {
```

Comment:
nice implementation! makes the code a lot more easier to read.

Reply:
Thanks - I figured that since there's a lot of fields in this struct I'd try something different.